src/index.js

```diff
@@ -0,0 +1,97 @@
+const { ScreepsAPI } = require("screeps-api");
+const Prometheus = require("prom-client");
+const config = require("./config/config.js");
+const http = require("http");
+
+Prometheus.collectDefaultMetrics({ timeout: 10000 });
+
+const api = new ScreepsAPI({
+  token: config.token,
+  protocol: "https",
+  hostname: "screeps.com",
+  port: 443,
+  path: "/", // Do not include '/api', it will be added automatically
+});
+
+const prometheusStats = {};
+
+function recursiveGauge(obj, prefix, labels) {
```

Comment:
I like this part, its a nice way to pull all the metrics from memory instead of hardcoding metric names both ingame and in the exporter.